spec/rubocop/cop/layout/empty_lines_around_arguments_spec.rb

```diff
@@ -121,6 +167,29 @@ def anything; end
       RUBY
       expect(cop.offenses.empty?).to be(true)
     end
+
+    it 'when trailing off of block' do
+      inspect_source(<<-RUBY.strip_indent)
+        fred.map do
+          <<-EOT
+            bar
+
+            foo
+          EOT
+        end.join("\n")
+      RUBY
+      expect(cop.offenses.empty?).to be(true)
+    end
+
+    it 'when method has no arguments' do
```

Comment:
or reworded to "it 'does not register an offense for a method with no arguments". I hope this makes sense to you.

Reply:
This does make more sense now. Thanks for your persistence!

My mindset with the `context` organization was to reduce text duplication. Dry descriptions I guess? 

This test file looks like this for `--format-documentation` output:

```
RuboCop::Cop::Layout::EmptyLinesAroundArguments
  autocorrects
    when empty line detected at bottom
    when args start on defintion line
    when empty line detected at top
    when empty line detected in the middle
    when multiple empty lines are detected
  does not register offense
    for multiple correctly listed mixed args
    for one line methods
    for correctly listed args starting on defintion line
    when trailing off of block
    when method has no arguments
    when passed block argument with empty line
  registers offense
    when multiple empty lines are detected
    when empty line detected at top
    when args start on defintion line
    when empty line detected at bottom
    when empty line detected in the middle
    when empty line before a block argument
    on correct lines
      for single offense example
      for multi-offense example
```

But this definitely conflicts with the principles in the Better Specs guide. And it also breaks from the standard we've established in the rest of the test suite. So a change is definitely warranted.  

I'll have an edit up shortly :).

Reply:
Here is the new look (randomized like the one above):

```
RuboCop::Cop::Layout::EmptyLinesAroundArguments
  when no extra lines
    accepts listed args starting on defintion line
    accepts block argument with empty line
    accepts multiple listed mixed args
    accpets one line methods
    accepts method with argument that trails off block
    accepts method with no arguments that trails off block
  when extra lines
    registers offense on correct line for single offense example
    registers offense for empty line between args
    registers offense when args start on definition line
    registers offense for empty line after arg
    registers offenses when multiple empty lines are detected
    autocorrects empty line detected at top
    registers offense on correct lines for multi-offense example
    registers offense for empty line in-between one arg
    registers offense when empty line between normal arg & block arg
    autocorrects multiple empty lines
    registers offense for empty line before arg
    autocorrects empty line in-between one arg
    autocorrects empty line detected at bottom
    autocorrects empty line detected in the middle
    autocorrects args that start on definition line
```

Reply:
👍